### PR TITLE
Mina-signer: Generate crypto constants from OCaml

### PR DESCRIPTION
**Pull request by mitschabaude**
Dump crypto constants from OCaml to an autogenerated JS file, when building snarkyjs

So far covers the following:
* Poseidon constants
* Hash prefixes
* base58 version bytes

### CODE REVIEW

File: scripts/build-snarkyjs-node.sh [hunk 1]
```diff
@@ -9,7 +9,8 @@ export DUNE_USE_DEFAULT_LINKER="y"
```
Comment (bkase) [inline scripts/build-snarkyjs-node.sh:9]: Can you add the typical prefix to the top of this so we catch all the errors:

```
#!/bin/bash

set -e
```

Then you can get rid of that `|| exit 1` and we'll catch more errors

**Review by bkase — APPROVED** [review]

File: scripts/build-snarkyjs-node.sh (hunk 1 above)
Comment (mitschabaude) [reply scripts/build-snarkyjs-node.sh:9]: oh this is what this does? cool!

**Review by mrmr1993 — APPROVED** [review]

**Comment by mitschabaude** [comment]
!ci-build-me

**Comment by mitschabaude** [comment]
!ci-build-me

**Comment by bkase** [comment]
!approved-for-mainnet

**Pull request merged**